### ISSUE TITLE
Table list query doesn't handle symlinks

### ISSUE THREAD

**Issue by puzpuzpuz** [questdb/questdb#2884]
### To reproduce

1. Create a table
2. Stop the database and move table dir to a different location on the same or a separate disk
3. Create a symlink in `root_dir/db/table_name` pointing to the new table dir location
4. Start the database and run `tables();`

### Expected Behavior

The table should be shown in the result set while it's not the case.

### Environment

```markdown
- **QuestDB version**: 6.6.1
- **OS**: Ubuntu 22.04, ext4
- **Browser**:
```

**Comment by marregui** [comment]
Hola!, but this would mean that the table is a symlink, what would that mean in terms of partition read only?

The rationale behind symlinks was to allow partitions to be "foreign". This issue would extend the domain of symlinks to include tables, which could have an impact on performance across all tables if the volume of a table is in very slow storage, across a network?.

**Comment by puzpuzpuz** [comment]
> what would that mean in terms of partition read only?

It's unrelated with partitions. The use case is moving a large table to a separate disk, nothing more. All partitions should remain writable. I guess some users might even want to move the entire `./db` directory to a separate disk.

> which could have an impact on performance across all tables if the volume of a table is in very slow storage, across a network?

Imagine that you have a table that requires 1TB of disk space while you have two disks 1.5TB each. In this situation you'd want to move the table to the second disk while all other tables remain on the first one. Performance won't be impacted as both disks are local.

**Comment by marregui** [comment]
cool, will do!

**Comment by marregui** [comment]
being addressed in https://github.com/questdb/questdb/pull/2710 there is a test to exemplify the description above `ServerMainForeignTableTest.testServerMainCreateTableMoveItsFolderAwayAndSoftLinkIt`.

**Comment by viper1** [comment]
> It's unrelated with partitions. The use case is moving a large table to a separate disk, nothing more. All partitions should remain writable. I guess some users might even want to move the entire `./db` directory to a separate disk.

thank you, this would be fantastic to have. 

a small extension with a new keyword CREATE ARCHIVE TABLE
that automatically creates the new table in archive_dir/db/table_name would make this completely seamless.

keep up the good work.

**Comment by marregui** [comment]
Hola @viper1 happy new year. Would you be able to elaborate on this last feature. Let's define it and see what can be done. Thank you in advance.

**Comment by viper1** [comment]
HNY @marregui. 

it is for the same use case from @puzpuzpuz above. for example we have two 1.5TB disks. 
the main db is on the first disk in root_dir. we create root_dir2 in the second disk. 
then we shut down the database, create the symlink in root_dir/db/table to point to root_dir2/db/table.

the extension is, instead of shutting down the database and manually creating the symlink:
1. define root_dir2 in the config. for example something like cairo.archive_dir=root_dir2/db
2. CREATE ARCHIVE TABLE understands that the table should be created in root_dir2 instead of root_dir
nothing more than that. the benefit is it allows the same functionality with no database downtime and no manual filesystem changes, especially if additional tables need to be created on the second disk

**Comment by marregui** [comment]
I suppose a more general solution would be to extend the create table statement so that it supports:

```sql
CREATE TABLE table_name ( ... ) PARTITION BY ... WITH .... IN VOLUME '/path/to/wherever/you/want'
```

so then questdb will create the table in '/path/to/wherever/you/want/table_name' and an automatic softlink "table_name" pointing there from the actual database root. Then no config is required. What are your thoughts?

**Comment by viper1** [comment]
this seems like an elegant solution - more general and also captures the use case well.

> CREATE TABLE table_name ( ... ) PARTITION BY ... WITH .... IN VOLUME '/path/to/wherever/you/want'

**Comment by marregui** [comment]
hi @viper1 I have pushed a commit https://github.com/questdb/questdb/pull/2710/commits/82e3dbf285774523d7b6cee6632e8d4c1dfed969 with the above requirement, it was very simple to do, that is why I did it. I will add more tests I suppose, and I will get feedback from @puzpuzpuz but it will be merged.

**Comment by viper1** [comment]
thank you, this will be great to have and keep up the good work!

**Comment by puzpuzpuz** [comment]
> I suppose a more general solution would be to extend the create table statement so that it supports:
> 
> ```sql
> CREATE TABLE table_name ( ... ) PARTITION BY ... WITH .... IN VOLUME '/path/to/wherever/you/want'
> ```
> 
> so then questdb will create the table in '/path/to/wherever/you/want/table_name' and an automatic softlink "table_name" pointing there from the actual database root

Such feature is insecure. Say, if a single server runs multiple QDB instances, this feature would allow anyone who has HTTP/PGWire access to one of the instances to read and modify any table belonging to other instances.

As a way to protect from that, we'd need to add a volume/path whitelist config option that would be empty by default.

**Comment by marregui** [comment]
good point, will add the config option, which will be a list of allowed paths, empty by default.

**Comment by viper1** [comment]
good point @puzpuzpuz 
agreed, empty whitelist default makes sense @marregui

**Comment by marregui** [comment]
`cairo.create.allowed.volume.paths` comma separated list of allowed paths.

@puzpuzpuz could you please review the PR, it should be almost ready for merging. Thank you in advance.

(https://github.com/questdb/questdb/pull/2710)

**Comment by puzpuzpuz** [comment]
> @puzpuzpuz could you please review the PR, it should be almost ready for merging. Thank you in advance.

Let wait until we merge #2752 as your PR has a lot of intersections with that one in terms of changed modules.

**Comment by marregui** [comment]
merge fest?

**Comment by marregui** [comment]
Hi @viper1 I hope you are well.

We have finally merged https://github.com/questdb/questdb/pull/2931 it has been quite a journey, I broke a lance for you. The PR description sort of explains how it works, and there is a bonus Dockerfile to set it up in a docker image.

This feature has not been released yet, it will in the upcoming release, in a week's time. It has been documented already https://questdb.io/docs/reference/sql/create-table/#table-target-volume .

Best regards,
